LibreNMS/Data/Store/Rrd.php

```diff
@@ -596,8 +597,11 @@ public function graph(string $options, array $env = null): string
         }
 
         // if valid image is returned with error, extract image and feedback
-        $image_type = Config::get('webui.graph_type', 'svg');
-        $search = $this->getImageEnd($image_type);
+        // rrdtool defaults to png if imgformat not specified
```

Comment:
This code feels like it could be improved.  It seems a bit fragile.

Reply:
perhaps we just search the string for both image ends...

Reply:
> This code feels like it could be improved. It seems a bit fragile.
> perhaps we just search the string for both image ends...

Yes, it felt a little fragile - sadly we don't have the original graph request parameters by this stage. Search for both image ends makes sense, I can make that change if preferrable.

I should also point out this doesn't resolve scenarios where a graph error needs to be generated in response to an rrd error for example, all those functions are still returning a graph in the image format of the system config, not in the same image format the graph was requested in.